core/src/main/java/com/datastax/oss/driver/api/core/config/OngoingConfigOptions.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core.config;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+
+/** An object where config options can be set programmatically. */
+public interface OngoingConfigOptions<SelfT extends OngoingConfigOptions<SelfT>> {
+
+  @NonNull
+  SelfT withBoolean(@NonNull DriverOption option, boolean value);
```

Comment:
Why don't you annotate these methods with `@CheckReturnValue`? This would be useful for `DriverExecutionProfile` implementations.

Reply:
See [this comment](https://github.com/datastax/java-driver/pull/1220#discussion_r271875069).